monitoring/metric_test.go

```diff
@@ -0,0 +1,66 @@
+package monitoring_test
+
+import (
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	. "github.com/cloudfoundry-community/splunk-firehose-nozzle/monitoring"
+	"github.com/cloudfoundry-community/splunk-firehose-nozzle/testing"
+	"github.com/cloudfoundry-community/splunk-firehose-nozzle/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Monitoring", func() {
+	var (
+		filtermetric string = "nozzle.queue.percentage,splunk.events.dropped.count,splunk.events.sent.count,firehose.events.dropped.count,firehose.events.received.count,splunk.events.throughput,nozzle.usage.ram,nozzle.usage.cpu,firehose.events.received.count.normalreceive,nozzle.cachehit.memory,nozzle.cachemiss.memory,nozzle.cachehit.remote,nozzle.cachemiss.remote,nozzle.cachehit.boltdb,nozzle.cachemiss.boltdb"
+		writer       testing.EventWriterMetricMock
+		monitor      = InitMetrics(lager.NewLogger("Test"), 2*time.Second, &writer, filtermetric)
+		Counter      utils.Counter
+	)
+
+	BeforeEach(func() {
+		RegisterFunc("nozzle.queue.percentage", func() interface{} { return 10 })
```

Comment:
use above created monitor to register func or counter instead of using global RegisterFunc